src/widget/core.rs

```diff
@@ -94,7 +95,7 @@ pub struct Pod {
 }
 
 #[derive(Debug)]
-pub(crate) struct WidgetState {
+pub struct WidgetState {
```

Comment:
Out of curiosity, why this change?

Reply:
This is to allow `set_origin` to be called with multiple different context types. As `set_origin` is public, and now takes `WidgetState` as a parameter, `WidgetState` also needs to be public.